File: 프로그래머스/unrated/135807. 숫자 카드 나누기/숫자 카드 나누기.py
def solve(a,b):
    while b!=0:
        a,b = b, a%b
    return a

def solution(arrayA, arrayB):
    
    a = arrayA[0]
    for i in range(len(arrayA)):
        a = solve(a, arrayA[i])
    
    b = arrayB[0]
    for i in range(len(arrayB)):
        b = solve(b, arrayB[i])
    
    for i in range(len(arrayA)):
        if arrayA[i]%b == 0:
            b = 1
        if arrayB[i]%a == 0:
            a = 1
    
    if a ==1 and b ==1 :
        return 0
    else:
        return max(a,b)
    

File: 프로그래머스/2/135807. 숫자 카드 나누기/숫자 카드 나누기.java
import java.util.*;
class Solution {
    public int solution(int[] arrayA, int[] arrayB) {
        int answer = 0;
        
        int ga = arrayA[0];
        int gb = arrayB[0];
        
        for(int i=1; i<arrayA.length; i++){
            ga = gcd(ga, arrayA[i]);
            gb = gcd(gb, arrayB[i]);
        }
        if(Dis(arrayA, gb)){
            if(answer < gb){
                answer = gb;
            }
        }
        if(Dis(arrayB, ga)){
            if(answer < ga){
                answer = ga;
            }
        }
        
        
        return answer;
    }
    static int gcd(int a, int b){
        if(b==0){
            return a;
        }
        return gcd(b, a%b);
    }
    static boolean Dis(int[] arr, int num){
        for(int n:arr){
            if(n%num==0) return false;
        }
        return true;
    }
}